### PR TITLE
Remove the BinaryCompatibility class as it is not useful on .NET Core…

### PR DESCRIPTION
**Pull request by AlexGhiondea**
… and creates issues on Debug builds when the TFM on the AppDomain is not recognized.

### CODE REVIEW

File: src/mscorlib/src/System/Globalization/DateTimeFormatInfo.cs [hunk 1]
```diff
@@ -2881,7 +2881,7 @@ private void AddMonthNames(TokenHashValue[] temp, String monthPostfix)
                                 // - if the app is targetting 4.5.1 or above OR the compat flag is set, use the correct behavior by default.
                                 // - if the app is targetting 4.5 or below AND the compat switch is set, use the correct behavior
                                 // - if the app is targetting 4.5 or below AND the compat switch is NOT set, use the incorrect behavior
-                                if (preferExistingTokens || BinaryCompatibility.TargetsAtLeast_Desktop_V4_5_1)
+                                if (preferExistingTokens)
```
Comment (akoeplinger) [inline src/mscorlib/src/System/Globalization/DateTimeFormatInfo.cs]: shouldn't this be the other way around? preferExistingTokens is always false on coreclr so you'd use the incorrect behavior in the else as per the comment above.

Comment (AlexGhiondea) [inline src/mscorlib/src/System/Globalization/DateTimeFormatInfo.cs]: You are right! We actually need to remove the other branch of the if and only keep the code (without the if) here.

**Comment by tarekgh** [comment]
>         }

if we are removing the check of preferExistingTokens below, should we just remove the whole initialization? as these will not be used at all?

---
Refers to: src/mscorlib/src/System/Globalization/DateTimeFormatInfo.cs:195 in 9d0b9b2. [](commit_id = 9d0b9b2e0913cb6201bb17cf3eeedebe6a89e508, deletion_comment = False)

**Comment by AlexGhiondea** [comment]
@tarekgh yup -- I can do that!

**Comment by tarekgh** [comment]
LGTM

**Comment by jkotas** [comment]
@dotnet-bot test OSX x64 Checked Build and Test  please

**Pull request merged**